internal/wire/header.go

```diff
@@ -32,59 +31,56 @@ type Header struct {
 	isPublicHeader bool
 }
 
-// ParseHeader parses the header.
-func ParseHeader(b *bytes.Reader, sentBy protocol.Perspective, version protocol.VersionNumber) (*Header, error) {
-	var typeByte uint8
-	if version == protocol.VersionUnknown {
-		var err error
-		typeByte, err = b.ReadByte()
-		if err != nil {
-			return nil, err
-		}
-		_ = b.UnreadByte() // unread the type byte
+// ParseHeaderFromServer parses the header for a packet that was sent by the server.
+func ParseHeaderFromServer(b *bytes.Reader, version protocol.VersionNumber) (*Header, error) {
```

Comment:
The name is ambiguous (and I got it wrong on my first reading). Either ParseHeaderSentByServer or ParseHeaderForClient.